base/number.jl

```diff
@@ -133,11 +133,41 @@ zero{T<:Number}(::Type{T}) = convert(T,0)
 """
     one(x)
 
-Get the multiplicative identity element for the type of `x` (`x` can also specify the type
-itself). For matrices, returns an identity matrix of the appropriate size and type.
+Return a multiplicative identity for `x` (which can be either a value
+or a type), of the same shape and numeric precision as `x`.  (For matrices,
```

Comment:
My perspective is that (a) `oneunit(x::T)` should be guaranteed to return something of type `T` and (b) `one` and `oneunit` should give the same result except for types representing dimensionful quantities (i.e. except for cases where the multiplicative identity *cannot* be the same type).